### PR TITLE
Changed file architecture to have collections of games, teams, and ga…

### PR DESCRIPTION
**Pull request by zachholcomb**
…meteams, removed that code from stat_tracker

We created collections of each class, game, game_team, team
Created corresponding tests for these classes as well

### CODE REVIEW

File: lib/game_collection.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'csv'
+
+class GameCollection
+  attr_reader :games_list
+
+  def initialize(file_path)
+    @games_list = create_games(file_path)
+  end
+
+  def create_games(file_path)
```
Comment (zachholcomb) [inline lib/game_collection.rb:10]: moved the csv reader to each collection class, the stat_tracker will pass the file_path to the game_collection here

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -5,35 +5,17 @@
 class StatTracker
   attr_reader :games, :teams, :game_teams
 
-  def initialize(games, teams, game_teams)
-    @games = games
-    @teams = teams
-    @game_teams = game_teams
+  def initialize(games_file, teams_, game_teams)
```
Comment (zachholcomb) [inline lib/stat_tracker.rb:8]: Changed the instance variables here to match what the collection classes are expecting, mostly for readability

File: lib/game_collection.rb (hunk 1 above)
Comment (StephanieFriend) [reply lib/game_collection.rb:10]: I like this setup. I think it will be easier to pull from

File: lib/game_team_collection.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+class GameTeamCollection
+  attr_reader :game_team_list
+
+  def initialize(file_path)
+    @game_team_list = create_game_teams(file_path)
+  end
+
+  def create_game_teams(file_path)
+    csv = CSV.read(file_path, headers: true, header_converters: :symbol)
+
```
Comment (StephanieFriend) [inline lib/game_team_collection.rb:12]: Maybe this will be a good chance for inheritance?

File: test/game_team_collection_test.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  def test_it_exists
+    assert_instance_of GameTeamCollection, @game_team_collection
+  end
+
+  def test_it_has_list_of_game_teams
+    assert_instance_of Array, @game_team_collection.game_team_list
+    assert_instance_of GameTeam, @game_team_collection.game_team_list.first
+    assert_equal 200, @game_team_collection.game_team_list.length
+  end
```
Comment (StephanieFriend) [inline test/game_team_collection_test.rb:19]: If we change the truncated files we'll have to make note to change this test

**Review by StephanieFriend — APPROVED** [review]
Overall I think these changes are great. Getting the CSV read's out of stattracker and into their own files I think will help us a lot when we get into further iterations. :)

**Pull request merged**